In [46]:
import pandas as pd
import numpy as np
import statistics as st


In [47]:
#data read
df=pd.read_csv(r"C:\Users\Kalraj Koteri\Desktop\puthon_practice\tips (1).csv")

In [48]:
df

total_bill   tip     sex smoker   day    time  size
0         16.99  1.01  Female     No   Sun  Dinner     2
1         10.34  1.66    Male     No   Sun  Dinner     3
2         21.01  3.50    Male     No   Sun  Dinner     3
3         23.68  3.31    Male     No   Sun  Dinner     2
4         24.59  3.61  Female     No   Sun  Dinner     4
..          ...   ...     ...    ...   ...     ...   ...
239       29.03  5.92    Male     No   Sat  Dinner     3
240       27.18  2.00  Female    Yes   Sat  Dinner     2
241       22.67  2.00    Male    Yes   Sat  Dinner     2
242       17.82  1.75    Male     No   Sat  Dinner     2
243       18.78  3.00  Female     No  Thur  Dinner     2

[244 rows x 7 columns]

In [49]:
# meta data 
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 244 entries, 0 to 243
Data columns (total 7 columns):
 #   Column      Non-Null Count  Dtype  
---  ------      --------------  -----  
 0   total_bill  244 non-null    float64
 1   tip         244 non-null    float64
 2   sex         244 non-null    object 
 3   smoker      244 non-null    object 
 4   day         244 non-null    object 
 5   time        244 non-null    object 
 6   size        244 non-null    int64  
dtypes: float64(2), int64(1), object(4)
memory usage: 13.5+ KB


In [50]:
#data cleaning 
#first we check data is missing or not 
#here any data is not missing
df.isnull().sum()

total_bill    0
tip           0
sex           0
smoker        0
day           0
time          0
size          0
dtype: int64

In [51]:
df.head()

total_bill   tip     sex smoker  day    time  size
0       16.99  1.01  Female     No  Sun  Dinner     2
1       10.34  1.66    Male     No  Sun  Dinner     3
2       21.01  3.50    Male     No  Sun  Dinner     3
3       23.68  3.31    Male     No  Sun  Dinner     2
4       24.59  3.61  Female     No  Sun  Dinner     4

In [52]:
df.drop_duplicates()

total_bill   tip     sex smoker   day    time  size
0         16.99  1.01  Female     No   Sun  Dinner     2
1         10.34  1.66    Male     No   Sun  Dinner     3
2         21.01  3.50    Male     No   Sun  Dinner     3
3         23.68  3.31    Male     No   Sun  Dinner     2
4         24.59  3.61  Female     No   Sun  Dinner     4
..          ...   ...     ...    ...   ...     ...   ...
239       29.03  5.92    Male     No   Sat  Dinner     3
240       27.18  2.00  Female    Yes   Sat  Dinner     2
241       22.67  2.00    Male    Yes   Sat  Dinner     2
242       17.82  1.75    Male     No   Sat  Dinner     2
243       18.78  3.00  Female     No  Thur  Dinner     2

[243 rows x 7 columns]

In [53]:
#motive to check how many day's in a week restaurant is open
df.day.value_counts()

day
Sat     87
Sun     76
Thur    62
Fri     19
Name: count, dtype: int64

only four day's in week rastaurant is opening

In [54]:
df.day.value_counts().index

Index(['Sat', 'Sun', 'Thur', 'Fri'], dtype='object', name='day')

In [55]:
df.day.value_counts().values

array([87, 76, 62, 19], dtype=int64)

In [56]:
#motive to find restaurant opened three time or not 
df.time

0      Dinner
1      Dinner
2      Dinner
3      Dinner
4      Dinner
        ...  
239    Dinner
240    Dinner
241    Dinner
242    Dinner
243    Dinner
Name: time, Length: 244, dtype: object

In [57]:
df.time.value_counts()

time
Dinner    176
Lunch      68
Name: count, dtype: int64

from above data the restaurant open only two time lunch and dinner

In [58]:
#motive to find the number of male and female came to restaurant for eating food
df.sex

0      Female
1        Male
2        Male
3        Male
4      Female
        ...  
239      Male
240    Female
241      Male
242      Male
243    Female
Name: sex, Length: 244, dtype: object

In [59]:
df.sex.value_counts()

sex
Male      157
Female     87
Name: count, dtype: int64

from above 157 male and 87 female came restaurant for eating food

In [60]:
df

total_bill   tip     sex smoker   day    time  size
0         16.99  1.01  Female     No   Sun  Dinner     2
1         10.34  1.66    Male     No   Sun  Dinner     3
2         21.01  3.50    Male     No   Sun  Dinner     3
3         23.68  3.31    Male     No   Sun  Dinner     2
4         24.59  3.61  Female     No   Sun  Dinner     4
..          ...   ...     ...    ...   ...     ...   ...
239       29.03  5.92    Male     No   Sat  Dinner     3
240       27.18  2.00  Female    Yes   Sat  Dinner     2
241       22.67  2.00    Male    Yes   Sat  Dinner     2
242       17.82  1.75    Male     No   Sat  Dinner     2
243       18.78  3.00  Female     No  Thur  Dinner     2

[244 rows x 7 columns]

In [61]:
#motive to find the total billing amount 
df.total_bill.sum()

4827.77

In [62]:
np.sum(df.total_bill)

4827.77

total billing amount of restaurant is 4827.77

In [63]:
#motive to find maximum bill  paying amount
df.total_bill.max()

50.81

In [64]:
np.max(df.total_bill)

50.81

maxmium bill paying amount is 50.81

In [32]:
#motive to find minimum paying bill amount
df.total_bill.min()

3.07

In [65]:
np.min(df.total_bill)

3.07

minimum  paying bill amount is 3.07

In [66]:
#motive to find the total tip amount
df.tip

0      1.01
1      1.66
2      3.50
3      3.31
4      3.61
       ... 
239    5.92
240    2.00
241    2.00
242    1.75
243    3.00
Name: tip, Length: 244, dtype: float64

In [67]:
df.tip.sum()

731.5799999999999

In [68]:
np.sum(df.tip)

731.5799999999999

tip amount of the restaurants is 731.5799999999999

In [69]:
#motive to find maximum tip amount
df.tip

0      1.01
1      1.66
2      3.50
3      3.31
4      3.61
       ... 
239    5.92
240    2.00
241    2.00
242    1.75
243    3.00
Name: tip, Length: 244, dtype: float64

In [70]:
df.tip.max()

10.0

In [71]:
np.max(df.tip)

10.0

max tip is 10.0

In [35]:
#motive to find minimum tip amount
df.tip.min()

1.0

In [72]:
np.min(df.tip)

1.0

minimum tips amount is 1.0

In [73]:
#motive to find average tip amount given by each customer 
df.tip.mean()

2.99827868852459

In [74]:
np.mean(df.tip)

2.99827868852459

In [76]:
st.mean(df.tip)

2.9982786885245902

average tip amount is 2.99827868852459

In [77]:
#motive to find total number of people came to restaurants
df['size']

0      2
1      3
2      3
3      2
4      4
      ..
239    3
240    2
241    2
242    2
243    2
Name: size, Length: 244, dtype: int64

In [78]:
df['size'].sum()

627

In [79]:
np.sum(df['size'])

627

total people came this restaurant is 627

In [82]:
#conditional
total_male=df[df.sex=='Male'].shape
total_male[0]

157

In [83]:
total_female=df[df.sex=='Female'].shape
total_female[0]

87

In [85]:
#motive to find total male smoker
df[(df.sex=="Male")&(df.smoker=="Yes")]

total_bill    tip   sex smoker   day    time  size
56        38.01   3.00  Male    Yes   Sat  Dinner     4
58        11.24   1.76  Male    Yes   Sat  Dinner     2
60        20.29   3.21  Male    Yes   Sat  Dinner     2
61        13.81   2.00  Male    Yes   Sat  Dinner     2
62        11.02   1.98  Male    Yes   Sat  Dinner     2
63        18.29   3.76  Male    Yes   Sat  Dinner     4
69        15.01   2.09  Male    Yes   Sat  Dinner     2
76        17.92   3.08  Male    Yes   Sat  Dinner     2
80        19.44   3.00  Male    Yes  Thur   Lunch     2
83        32.68   5.00  Male    Yes  Thur   Lunch     2
90        28.97   3.00  Male    Yes   Fri  Dinner     2
95        40.17   4.73  Male    Yes   Fri  Dinner     4
96        27.28   4.00  Male    Yes   Fri  Dinner     2
97        12.03   1.50  Male    Yes   Fri  Dinner     2
98        21.01   3.00  Male    Yes   Fri  Dinner     2
105       15.36   1.64  Male    Yes   Sat  Dinner     2
106       20.49   4.06  Male    Yes   Sat  Dinner     2
107       25.21   4.29  Male    Yes   Sat  Dinner     2
138       16.00   2.00  Male    Yes  Thur   Lunch     2
170       50.81  10.00  Male    Yes   Sat  Dinner     3
171       15.81   3.16  Male    Yes   Sat  Dinner     2
172        7.25   5.15  Male    Yes   Sun  Dinner     2
173       31.85   3.18  Male    Yes   Sun  Dinner     2
174       16.82   4.00  Male    Yes   Sun  Dinner     2
175       32.90   3.11  Male    Yes   Sun  Dinner     2
176       17.89   2.00  Male    Yes   Sun  Dinner     2
177       14.48   2.00  Male    Yes   Sun  Dinner     2
179       34.63   3.55  Male    Yes   Sun  Dinner     2
180       34.65   3.68  Male    Yes   Sun  Dinner     4
181       23.33   5.65  Male    Yes   Sun  Dinner     2
182       45.35   3.50  Male    Yes   Sun  Dinner     3
183       23.17   6.50  Male    Yes   Sun  Dinner     4
184       40.55   3.00  Male    Yes   Sun  Dinner     2
187       30.46   2.00  Male    Yes   Sun  Dinner     5
189       23.10   4.00  Male    Yes   Sun  Dinner     3
190       15.69   1.50  Male    Yes   Sun  Dinner     2
192       28.44   2.56  Male    Yes  Thur   Lunch     2
193       15.48   2.02  Male    Yes  Thur   Lunch     2
194       16.58   4.00  Male    Yes  Thur   Lunch     2
196       10.34   2.00  Male    Yes  Thur   Lunch     2
199       13.51   2.00  Male    Yes  Thur   Lunch     2
200       18.71   4.00  Male    Yes  Thur   Lunch     3
204       20.53   4.00  Male    Yes  Thur   Lunch     4
206       26.59   3.41  Male    Yes   Sat  Dinner     3
207       38.73   3.00  Male    Yes   Sat  Dinner     4
208       24.27   2.03  Male    Yes   Sat  Dinner     2
210       30.06   2.00  Male    Yes   Sat  Dinner     3
211       25.89   5.16  Male    Yes   Sat  Dinner     4
216       28.15   3.00  Male    Yes   Sat  Dinner     5
217       11.59   1.50  Male    Yes   Sat  Dinner     2
218        7.74   1.44  Male    Yes   Sat  Dinner     2
220       12.16   2.20  Male    Yes   Fri   Lunch     2
222        8.58   1.92  Male    Yes   Fri   Lunch     1
224       13.42   1.58  Male    Yes   Fri   Lunch     2
230       24.01   2.00  Male    Yes   Sat  Dinner     4
231       15.69   3.00  Male    Yes   Sat  Dinner     3
234       15.53   3.00  Male    Yes   Sat  Dinner     2
236       12.60   1.00  Male    Yes   Sat  Dinner     2
237       32.83   1.17  Male    Yes   Sat  Dinner     2
241       22.67   2.00  Male    Yes   Sat  Dinner     2

In [86]:
male_smoker=df[(df.sex=="Male")&(df.smoker=="Yes")].shape
male_smoker[0]

60

In [87]:
No_smoker=df[(df.sex=="Male")&(df.smoker=="No")].shape
No_smoker[0]

97

In [88]:
print("percentage of no male smoker :",(No_smoker[0]/157)*100)

percentage of no male smoker : 61.78343949044586


In [89]:
print("percentage of male smoker :",(male_smoker[0]/total_male[0])*100)

percentage of male smoker : 38.21656050955414


In [91]:
total_female=df[df.sex=="Female"].value_counts().sum()
total_female

87

In [92]:
female_smoker=df[(df.sex=='Female')&(df.smoker=="Yes")].shape
female_smoker[0]

33

In [93]:
print("percentage of female smoker :",(female_smoker[0]/total_female)*100)


percentage of female smoker : 37.93103448275862


In [94]:
No_female_smoker=df[(df.sex=="Female")&(df.smoker=="No")].shape
No_female_smoker[0]

54

In [95]:
print("percentage of no female smoker :",(No_female_smoker[0]/total_female)*100)


percentage of no female smoker : 62.06896551724138


#report
1>--percentage of no female smoker : 62.06896551724138
2>--percentage of female smoker : 37.93103448275862
3>--percentage of no male smoker : 61.78343949044586
4>--percentage of male smoker : 38.21656050955414

In [96]:
max_tip=max(df.tip)
max_tip

10.0

In [97]:
df[(df.tip==max_tip)]

total_bill   tip   sex smoker  day    time  size
170       50.81  10.0  Male    Yes  Sat  Dinner     3

In [98]:
min_tip=min(df.tip)
min_tip

1.0

In [99]:
df[df.tip==min_tip]

total_bill  tip     sex smoker  day    time  size
67         3.07  1.0  Female    Yes  Sat  Dinner     1
92         5.75  1.0  Female    Yes  Fri  Dinner     2
111        7.25  1.0  Female     No  Sat  Dinner     1
236       12.60  1.0    Male    Yes  Sat  Dinner     2

In [100]:
df[(df.tip==min_tip)&(df.sex=="Female")&(df.time=="Dinner")].value_counts().sum()


3

#reprot
max tip given by male and it came at dinner time and he was a smoker

In [101]:
day_wise=df.day.value_counts()
day_wise

day
Sat     87
Sun     76
Thur    62
Fri     19
Name: count, dtype: int64

In [102]:
for i in range(len(day_wise)):
    print("day wise percentage of people vist to the restaurant :",i,(day_wise[i]/244)*100)

day wise percentage of people vist to the restaurant : 0 35.65573770491803
day wise percentage of people vist to the restaurant : 1 31.147540983606557
day wise percentage of people vist to the restaurant : 2 25.40983606557377
day wise percentage of people vist to the restaurant : 3 7.786885245901639


#report
restaurant open 4 days in a week
most of the people visit the restaurant on saturday

In [103]:
df[df.sex=="Male"].tip.sum()

485.07

In [105]:
df[df.sex=="Male"].tip.max()

10.0

In [112]:
df[df.sex=="Male"].tip.min()

1.0

In [113]:
df[df.sex=="Male"].time.value_counts()

time
Dinner    124
Lunch      33
Name: count, dtype: int64

In [114]:
df[df.sex=="Female"].tip.sum()

246.50999999999996

In [115]:
df[df.sex=="Female"].tip.max()

6.5

In [116]:
df[df.sex=="Female"].time.value_counts()

time
Dinner    52
Lunch     35
Name: count, dtype: int64

In [ ]:
df[df.sex=="Female"].tip.min()